contracts/exchange/contracts/src/MixinSignatureValidator.sol

```diff
@@ -244,32 +249,26 @@ contract MixinSignatureValidator is
             hash,
             signature
         );
+        bool success;
         assembly {
             let cdStart := add(callData, 32)
-            let success := staticcall(
+            success := staticcall(
                 gas,              // forward all gas
                 walletAddress,    // address of Wallet contract
                 cdStart,          // pointer to start of input
                 mload(callData),  // length of input
                 cdStart,          // write output over input
                 32                // output size is 32 bytes
             )
-
-            switch success
-            case 0 {
-                // Revert with `Error("WALLET_ERROR")`
-                mstore(0, 0x08c379a000000000000000000000000000000000000000000000000000000000)
-                mstore(32, 0x0000002000000000000000000000000000000000000000000000000000000000)
-                mstore(64, 0x0000000c57414c4c45545f4552524f5200000000000000000000000000000000)
-                mstore(96, 0)
-                revert(0, 100)
-            }
-            case 1 {
+            if eq(success, 1) {
                 // Signature is valid if call did not revert and returned true
                 isValid := mload(cdStart)
             }
         }
-        return isValid;
+        if (success)
+            return isValid;
+        rrevert(
+            SignatureError(hash, SignatureErrorCodes.WALLET_ERROR));
```

Comment:
This feels a bit unorthodox to me. Can we do:

```
if (!success) {
            rrevert(SignatureError(hash, SignatureErrorCodes.WALLET_ERROR));
}
return isValid;
```

Another nit: it would be nice if boolean naming conventions followed the rest of our codebase. Perhaps `didSucceed`?